tests/network/bindingplugin_macvtap.go

```diff
@@ -0,0 +1,260 @@
+		nodeList := libnode.GetAllSchedulableNodes(kubevirt.Client())
+		Expect(nodeList.Items).NotTo(BeEmpty(), "schedulable kubernetes nodes must be present")
+		nodeName := nodeList.Items[0].Name
+
+		chosenMACHW, err := GenerateRandomMac()
+		Expect(err).ToNot(HaveOccurred())
+		chosenMAC := chosenMACHW.String()
+
+		serverCIDR := "192.0.2.102/24"
+		serverIP, err := libnet.CidrToIP(serverCIDR)
+		Expect(err).ToNot(HaveOccurred())
+
+		_ = createAlpineVMIStaticIPOnNode(nodeName, macvtapNetworkName, "eth0", serverCIDR, &chosenMAC)
+		clientVMI := createAlpineVMIStaticIPOnNode(nodeName, macvtapNetworkName, "eth0", "192.0.2.101/24", nil)
+
+		Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed())
+	})
+
+	Context("VMI migration", func() {
+		var clientVMI *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			checks.SkipIfMigrationIsNotPossible()
+		})
+
+		BeforeEach(func() {
+			macAddressHW, err := GenerateRandomMac()
+			Expect(err).ToNot(HaveOccurred())
+			macAddress := macAddressHW.String()
+			clientVMI, err = createAlpineVMIRandomNode(macvtapNetworkName, macAddress)
+			Expect(err).NotTo(HaveOccurred(), "must succeed creating a VMI on a random node")
+		})
+
+		It("should be successful when the VMI MAC address is defined in its spec", func() {
+			By("starting the migration")
+			migration := libmigration.New(clientVMI.Name, clientVMI.Namespace)
+			migration = libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(kubevirt.Client(), migration)
+
+			// check VMI, confirm migration state
+			libmigration.ConfirmVMIPostMigration(kubevirt.Client(), clientVMI, migration)
+		})
+
+		Context("with live traffic", func() {
+			var serverVMI *v1.VirtualMachineInstance
+			var serverVMIPodName string
+			var serverIP string
+
+			macvtapIfaceIPReportTimeout := 4 * time.Minute
+
+			waitVMMacvtapIfaceIPReport := func(vmi *v1.VirtualMachineInstance, macAddress string, timeout time.Duration) (string, error) {
+				var vmiIP string
+				err := wait.PollImmediate(time.Second, timeout, func() (done bool, err error) {
+					vmi, err := kubevirt.Client().VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, &k8smetav1.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+
+					for _, iface := range vmi.Status.Interfaces {
+						if iface.MAC == macAddress {
+							if ip := iface.IP; ip != "" {
+								vmiIP = ip
+								return true, nil
+							}
+							return false, nil
+						}
+					}
+
+					return false, nil
+				})
+				if err != nil {
+					return "", err
+				}
+
+				return vmiIP, nil
+			}
+
+			waitForPodCompleted := func(podNamespace string, podName string) error {
+				pod, err := kubevirt.Client().CoreV1().Pods(podNamespace).Get(context.TODO(), podName, k8smetav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				if pod.Status.Phase == k8sv1.PodSucceeded || pod.Status.Phase == k8sv1.PodFailed {
```

Comment:
How come `pod.Status.Phase == k8sv1.PodFailed` is a reasonable option?

Reply:
I guess this check exist to not make the test fail if virt-launcher shut down ungracefully due to some bug or unexpected issues on CI.

Reply:
To me it doesn't make sense to expect such failure in the test. But since this is not the focus of the test (it is just a way to avoid a race) and not the focus of this PR, let's finalize the discussion. I won't block the PR if you keep it.

Reply:
Please let me know if you like me to post a follow up to address it, thank you :slightly_smiling_face: